relay-server/src/services/processor.rs

```diff
@@ -1129,6 +1128,12 @@ impl EnvelopeProcessorService {
         &self,
         state: &mut ProcessEnvelopeState<G>,
     ) -> Result<(), ProcessingError> {
+        let project_state = Arc::clone(&state.project_state);
```

Comment:
Do we need the clone here?

Reply:
In the current setup, yes, because we need mutable reference to `ProcessEnvelopeState` to mutate the envelope, and the rules are inside the project state, so we'd get a `cannot borrow mutable because its also borrowed as immutable` error